python/hail/context.py

```diff
@@ -365,7 +365,13 @@ def read(self, path, sites_only=False, samples_only=False):
         :rtype: :class:`.VariantDataset`
         """
 
-        jvds = self._jhc.readAll(jindexed_seq_args(path), sites_only, samples_only)
+        is_generic_genotype = self._jhc.readAllMetadata(jindexed_seq_args(path)).isGenericGenotype()
```

Comment:
I don't like read_generic, because it's really not obvious to a user of the python interface whether they've got a GDS or VDS.